bob-common/src/node.rs

```diff
@@ -82,12 +82,12 @@ impl Node {
         self.conn.read().await.clone()
     }
 
-    pub async fn check(&self, client_fatory: &Factory) -> Result<(), String> {
+    pub async fn check(&self, client_factory: &Factory, local_node_name: &String) -> Result<(), String> {
```

Comment:
This change seems to break the idea of factory pattern. `local_node_name` should be part of `Factory`, not the separate argument